src/app/page.tsx

```diff
@@ -1,21 +1,85 @@
-import { getTopStories } from "@/services/hn";
-import Link from "next/link";
+'use client';
+import { getTopStories } from '@/services/hn';
+import Link from 'next/link';
+import { useRef, useEffect, useState } from 'react';
 
-export default async function Home() {
-  const topstories = await getTopStories();
+export default function Home() {
+  const [start, setStart] = useState(0);
+  const limit = 10;
+  const [topstories, setTopStories] = useState([]);
```

Comment:
Missing a type here so when I try to build i get an error.
```suggestion
  const [topstories, setTopStories] = useState<Story[]>([]);
```